libcasr/src/js.rs

```diff
@@ -0,0 +1,379 @@
+//! JS module implements `ParseStacktrace` and `Exception` traits for JS reports.
+use crate::error::{Error, Result};
+use crate::exception::Exception;
+use crate::execution_class::ExecutionClass;
+use crate::stacktrace::{ParseStacktrace, Stacktrace, StacktraceEntry};
+
+use regex::Regex;
+
+/// Structure provides an interface for parsing JS exception message.
+pub struct JsException;
+
+impl Exception for JsException {
+    fn parse_exception(stderr: &str) -> Option<ExecutionClass> {
+        let rexception = Regex::new(r"^.*?(\S*Error):(?:\s+(.*))?$").unwrap();
```

Comment:
What about warning exceptions like `Warning: unreachable code after return statement`

Reply:
Warnings cannot be exceptions, they can only be displayed via `console.warn` (unless they are custom `Warning` objects). It seems that the logic is about handling real exceptions, so warnings don't need to be handled here.